tests/test_argument_check.py

```diff
@@ -189,10 +186,19 @@ def test_missing_params_correct_error(test_type, module, field, vcs_type, error_
     # However, this doesn't work if the chosen setting is equal to the one passed in parameters. In that case we use
     # some other setting.
     if test_type == "main":
-        if module == "Launcher" and field == "config_path":
-            settings.Vcs.type = ""
+        if module == "Vcs" and field == "type":
+            return
+        elif module == "LocalMainVcs" and field == "source_dir":
+            return
+        elif module == "GitVcs" and field == "refspec" and settings.Vcs.type == "gerrit":
+            return
+        elif module in ["PerforceVcs", "PerforceMainVcs"]:
+            if field == "port":
+                settings.PerforceVcs.user = ""
+            else:
+                settings.PerforceVcs.port = ""
         else:
-            settings.Launcher.config_path = ""
+            settings.Vcs.type = ""
```

Comment:
This line prevents vcs errors from happening.
Therefore there is no chance to check if vcs errors are produced even if vcs parameters are properly specified.